### ISSUE TITLE
Refactor environment variables for services (ENABLE_.../DISABLE_...)

### ISSUE THREAD

**Issue by Josef-Friedrich** [tomav/docker-mailserver#297]
All services should be disabled by default to let users configure their container.
# Current situation:

```
environment:
    - DISABLE_AMAVIS=1
    - DISABLE_CLAMAV=1
    - DISABLE_SPAMASSASSIN=1
    - ENABLE_FAIL2BAN=1
    - ENABLE_FETCHMAIL=1
    - ENABLE_MANAGESIEVE=1
    - ENABLE_POP3=1
```
# Proposal 1:

```
    - ENABLE_AMAVIS=1
    - ENABLE_CLAMAV=1
    - ENABLE_FAIL2BAN=1
    - ENABLE_FETCHMAIL=1
    - ENABLE_MANAGESIEVE=1
    - ENABLE_POP3=1
    - ENABLE_SPAMASSASSIN=1
```
# Proposal 2:

```
environment:
    - SERVICES="amavis clamav fail2ban fetchmail managesieve pop3 spamassassin"
```
# Proposal 3:

Proposal 1 + Proposal 2

**Comment by tomav** [comment]
Hi @Josef-Friedrich, I was thinking about proposal 1 but 2 also works.
What's you opinion?

**Comment by Josef-Friedrich** [comment]
My favourite is `Proposal 3`. Proposal 3 is probably a little bit harder to implement, but gives the user more control over how they want configure `docker-mailserver`.

**Comment by tomav** [comment]
Mine goes to proposal 2 because it's more explicit and meaningful.
Perhaps a bit easier to implement test on the env matrix too.

**Comment by Josef-Friedrich** [comment]
Ok ... I will try proposal 2

```
environment:
    - SERVICES="amavis clamav fail2ban fetchmail managesieve pop3 spamassassin"
```

**Comment by tomav** [comment]
Sorry... typo in my previous post. Was talking about proposal 1. Really sorry.
Perhaps we should ask for @KyleOndy advice.

**Comment by KyleOndy** [comment]
I prefer Proposal 1.

Since all this config should be stored within a script or a ~docker-compose.yml` the verbosity should not be a concern.

Checking a single environmental variable, as we do now in all the scripts, for a service seems cleaner than checking for the existence for the service name in a single environmental variable.

**Comment by tomav** [comment]
Ok for me.

**Comment by Josef-Friedrich** [comment]
Ok! I’ll try to implement proposal 1.

**Comment by kachkaev** [comment]
Looks like this is a breaking change, so needs to become v3, not 2.1 Current users should be able to keep pulling `tvial/docker-mailserver:v2` without changing ENV.

Awesome stuff BTW! Really looking for it, mainly because my container is using ≈0.5G of memory atm.

**Comment by kachkaev** [comment]
@Josef-Friedrich would you still like to have a look at this? Just wondering what your plans are.

**Comment by Josef-Friedrich** [comment]
Yes, but at the moment I'm busy

**Comment by tomav** [comment]
Fixed by #432
